crates/rolldown_binding/src/types/binding_outputs.rs

```diff
@@ -1,29 +1,40 @@
+use std::{
+  cell::RefCell,
+  sync::{Arc, Mutex},
+  vec,
+};
+
 use napi_derive::napi;
 
 use super::{binding_output_asset::BindingOutputAsset, binding_output_chunk::BindingOutputChunk};
 
-/// The `BindingOutputs` owner `Vec<Output>` the mutable reference, it avoid `Clone` at call `writeBundle/generateBundle` hook, and make it mutable.
 #[napi]
 pub struct BindingOutputs {
-  inner: &'static mut Vec<rolldown_common::Output>,
+  inner: Arc<Mutex<RefCell<Vec<rolldown_common::Output>>>>,
```

Comment:
Is the `RefCell` necessary? There is already a `Mutex`.

Reply:
It is easy to muate `RefCell<T>` than `Arc<T>`.

Reply:
Can you give an example? I don't see benefits.